hmpps-prisoner-search-indexer/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/services/PrisonerSynchroniserService.kt

```diff
@@ -116,19 +115,22 @@ class PrisonerSynchroniserService(
       }
       isUpdated
     } ?: run {
-      // Need to create, which means calling all domain data too
+      // Need to create, which means calling all domain data too.
+      // NOTE if multiple messages are received for the same prisoner at the same time
+      // the create operation could be attempted multiple times
 
       val prisoner = Prisoner().translate(
         existingPrisoner = null,
         ob = ob,
         incentiveLevel = runCatching { getIncentive(ob) },
         restrictedPatientData = runCatching { getRestrictedPatient(ob) },
       )
-      prisonerRepository.save(prisoner, SyncIndex.RED)
+      prisonerRepository.createPrisoner(prisoner, SyncIndex.RED)
+      // If prisoner already exists in opensearch, an exception is thrown (same as for version conflict with update)
```

Comment:
So any multiple attempts to create, due to multiple sqs events will fail with an exception casing a retry. Which will then do an update instead and _probably_ result in a no-op.